ThunderstoreCLI/Utils/Spinner.cs

```diff
@@ -23,24 +21,37 @@ public ProgressSpinner(string label, Task[] tasks)
 
     public async Task Start()
     {
+        // Cursor operations are not always available e.g. in GitHub Actions environment. Done up here to minimize exception usage (throws and catches are expensive and all)
+        bool canUseCursor;
+        try
+        {
+            // nop that will throw if cursor position can't be gotten
+            _ = Console.CursorTop;
+            canUseCursor = true;
+        }
+        catch
+        {
+            canUseCursor = false;
+        }
+
         while (true)
         {
-            if (_tasks.Any(x => x.IsFaulted))
+            IEnumerable<Task> faultedTasks;
+            if ((faultedTasks = _tasks.Where(static x => x.IsFaulted)).Any())
             {
                 Write.Empty();
-                throw new SpinnerException("Some of the tasks have faulted");
+                throw new SpinnerException("Some of the tasks have faulted", faultedTasks.Select(x => x.Exception!));
             }
 
             var completed = _tasks.Count(static x => x.IsCompleted);
-            var spinner = completed == _tasks.Length ? "✓" : _spinChars[_spinIndex++ % _spinChars.Length];
 
-            // Cursor operations are not always available e.g. in GitHub Actions environment.
-            try
+            if (canUseCursor)
             {
+                var spinner = completed == _tasks.Length ? '✓' : _spinChars[_spinIndex++ % _spinChars.Length];
                 Console.SetCursorPosition(0, Console.CursorTop);
-                Console.Write(Green($"{completed}/{_tasks.Length} {_label} {spinner}"));
+                Write.Success($"{completed}/{_tasks.Length} {_label} {spinner}");
```

Comment:
IIRC `Write.Success` uses `Console.WriteLine` under the hood, does switching away from `Console.Write` cause the spinner animation to be shown on multiple lines?